### PR TITLE
Extra issue fix

### PR DESCRIPTION
**Pull request by sansx**
1. optimize TonConnectUIProvider

2. use Symble key in provide / inject
```
// now we use tonConnectUIKey export from @townsquarexyz/ui-vue to inject instance in demo project, like this usage
<script  lang="ts">
import { TonConnectUI, useTonWallet, tonConnectUIKey } from "@townsquarexyz/ui-vue";
const tonConnectUI = inject<TonConnectUI | null>(tonConnectUIKey, null);
</script>
```

which related to #8 [Moderate] implementation 1, 2, 3, 4

### CODE REVIEW

File: src/components/TonConnectUIProvider.vue [hunk 1]
```diff
@@ -1,8 +1,9 @@
 <script lang="ts">
```
Comment (dnischeta) [inline src/components/TonConnectUIProvider.vue:1]: Wait, provider is implemented incorrect. I see, why.

In the issue I put next example of code:

```
// <TonConnectUIProvider />
const tonConnectUI = shallowRef<TonConnectUI | null>(null);

provide('ton-connect-ui', tonConnectUI);

// Injection consumer, for example `useTonWallet`
const tonConnectUi = inject('ton-connect-ui'); // That line of code should be inside `useTonConnectUI`, put it here for simplification
const unsub = shallowRef<() => void>(() => {});
const wallet = shallowRef<...>(null);

onMounted(() => {
  watch(tonConnectUI, (actualTonConnectUI) => {
    unsub(); // Unsubscribe on watcher trigger <-> when tonConnectUI instance is updated (as done in React)

    wallet.value = actualTonConnectUI.wallet;

    unsub.value = actualTonConnectUI.onStatusChange((value) => {
      wallet.value = value;
    });
  });
});

onUnmounted(() => {
  unsub.value();
});
```

Pay attention to the comment `// Injection consumer, for example `useTonWallet``. So all next code should be placed into `useTonWallet`.

File: src/components/TonConnectUIProvider.vue [hunk 2]
```diff
@@ -39,7 +40,7 @@ export default defineComponent({
     );
 
     // Provide the TonConnectUI instance to child components
-    provide("tonConnectUI", tonConnectUI.value);
+    provide(tonConnectUIKey, tonConnectUI.value);
```
Comment (dnischeta) [inline src/components/TonConnectUIProvider.vue:43]: Actually, as I see above, `tonConnectUI.value.uiOptions` are updated using watch. So instance of TonConnectUI is the same. Therefore, you don't need to subscribe for `tonConnectUI.value` updates in consumers (eg. useTonWallet).

So just remove useless code from that component (`wallet`, `unsubscribe`, etc.)

**Review by dnischeta — CHANGES_REQUESTED** [review]

**Review by dnischeta — APPROVED** [review]
I suppose, you can move forward to the testing and preparing for the release.

File: README.md [hunk 3]
```diff
@@ -82,8 +91,8 @@ export default {
 </script>
 ```
 
-You can add `class` and `:style` props to the button as well. Note that you cannot pass child to the TonConnectButton. 
-`<TonConnectButton class="my-button-class" :style="{ float: 'right' }"/>`
+You can add `buttonRootId` props to the button as well.
+`<TonConnectButton :button-root-id="`ton-connect-button`" />`
```
[the remark this answers is not in the document]
Comment (sansx) [reply README.md]: done

**Comment by dnischeta** [comment]
> feel free to notify Alex that the repo is ready to merge

Am I entitled to do it? I mean, he asked me to do a general review. We can clear everything up in Telegram.

> Also, could we connect on Telegram for smoother communication?

Feel free to contact on Telegram: https://t.me/dimchickLoveNY .

**Comment by sansx** [comment]
Yeah, since Alex asked you to handle the general review, we’d like to let you inform him to keep the process smooth and complete, and then we can push this into next step. And it's not the must, we can let him know this in some other day. :D

**Pull request merged**